File: Projects/rosalind/MPRT.java
import java.io.*;
import java.util.*;
import java.util.regex.*;
import java.util.stream.*;
import java.net.*;

public class MPRT {
	private static final Pattern pattern = Pattern.compile("(?=N[^P][ST][^P])");
	public static void main(String args[]) throws IOException {
		Scanner in = new Scanner(System.in);
		PrintStream stdout = System.out;
		while (in.hasNext()) {
			String id = in.next(), line = null;
			StringBuilder seq = new StringBuilder();
			URL url = new URL("http://www.uniprot.org/uniprot/" + id + ".fasta");
			Scanner urlin = new Scanner(url.openStream());
			urlin.nextLine();
			while (urlin.hasNext())
				seq.append(urlin.next());
			Matcher matcher = pattern.matcher(seq);
			List<Integer> list = new ArrayList<>();
			while (matcher.find())
				list.add(matcher.start() + 1);
			if (list.size() != 0) {
				stdout.println(id);
				stdout.println(list.stream().map(String::valueOf).collect(Collectors.joining(" ")));
			}
			urlin.close();
		}
		in.close();
	}
}

File: Projects/rosalind/MPRT.kt
package rosalind

import java.io.IOException
import java.net.URL
import java.util.*
import java.util.regex.Pattern
import java.util.stream.Collectors

object MPRT {
    private val pattern = Pattern.compile("(?=N[^P][ST][^P])")
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        val stdout = System.out
        while (stdin.hasNext()) {
            val id = stdin.next()
            val seq = StringBuilder()
            val url = URL("http://www.uniprot.org/uniprot/$id.fasta")
            val urlin = Scanner(url.openStream())
            urlin.nextLine()
            while (urlin.hasNext()) seq.append(urlin.next())
            val matcher = pattern.matcher(seq)
            val list: MutableList<Int> = ArrayList()
            while (matcher.find()) list.add(matcher.start() + 1)
            if (list.size != 0) {
                stdout.println(id)
                stdout.println(list.stream().map { obj: Int -> obj.toString() }.collect(Collectors.joining(" ")))
            }
            urlin.close()
        }
        stdin.close()
    }
}